relayer/src/tx/client.rs

```diff
@@ -11,19 +11,32 @@ use ibc::tx_msg::Msg;
 use crate::chain::{Chain, CosmosSDKChain};
 use crate::config::ChainConfig;
 use crate::error::{Error, Kind};
+use crate::keyring::store::{KeyEntry, KeyRingOperations};
 use crate::util::block_on;
+use bitcoin::hashes::hex::ToHex;
 use ibc::ics02_client::height::Height;
+use std::str::FromStr;
+use tendermint::account::Id as AccountId;
 
 #[derive(Clone, Debug)]
 pub struct CreateClientOptions {
     pub dest_client_id: ClientId,
     pub dest_chain_config: ChainConfig,
     pub src_chain_config: ChainConfig,
+    pub signer_key: String,
 }
 
-pub fn create_client(opts: CreateClientOptions) -> Result<(), Error> {
+pub fn create_client(opts: CreateClientOptions) -> Result<Vec<u8>, Error> {
     // Get the destination chain
-    let dest_chain = CosmosSDKChain::from_config(opts.clone().dest_chain_config)?;
+    let mut dest_chain = CosmosSDKChain::from_config(opts.clone().dest_chain_config)?;
+
+    // Get the key from key seed file
+    let key = dest_chain
+        .keybase
+        .key_from_seed_file(&opts.signer_key)
+        .map_err(|e| Kind::KeyBase.context(e))?;
+    let signer: AccountId =
+        AccountId::from_str(&key.address.to_hex()).map_err(|e| Kind::KeyBase.context(e))?;
```

Comment:
I think the idea to abstract this to the chain is the right direction but maybe we should brainstorm where this belongs to, there's also a context, keybase, etc. Maybe we should include this as part of the refactor instead of doing one way now and deciding later on another approach and have to refactor it again. Since we only have two txs I'd keep like this for now and when we decide the final architecture I can refactor this. Does it make sense?

Reply:
Yes agreed.